packages/editor/src/components/post-author/index.js

```diff
@@ -16,34 +23,145 @@ export class PostAuthor extends Component {
+		if ( query.length < 2 ) {
+			return;
+		}
+
+		if ( this.searchCache[ query ] ) {
+			populateResults( this.resolveResults( this.searchCache[ query ] ) );
+			return;
+		}
+
+		this.requestResults( query, populateResults );
 	}
 
-	setAuthorId( event ) {
+	resolveResults( results ) {
+		this.authors = results;
+		return results.map( ( author ) => ( author.name ) );
+	}
+
+	getCurrentAuthor( authorId ) {
+		if ( ! authorId ) {
+			return;
+		}
+		apiFetch( { path: `/wp/v2/users/${ encodeURIComponent( authorId ) }` } ).then( ( results ) => {
+			this.setState( { postAuthor: results } );
+		} );
+	}
+
+	setAuthorId( selection ) {
+		if ( ! selection ) {
+			return;
+		}
 		const { onUpdateAuthor } = this.props;
-		const { value } = event.target;
-		onUpdateAuthor( Number( value ) );
+		if ( typeof selection === 'string' ) {
+			// Author name from the autocompleter.
+			const author = this.authors.find( ( singleAuthor ) => {
+				return singleAuthor.name === selection;
+			} );
+			onUpdateAuthor( Number( author.id ) );
+		} else {
+			// Author ID from the select.
+			onUpdateAuthor( Number( selection.target.value ) );
+		}
 	}
 
 	render() {
-		const { postAuthor, instanceId, authors } = this.props;
-		const selectId = 'post-author-selector-' + instanceId;
-
-		// Disable reason: A select with an onchange throws a warning
+		const { postAuthor } = this.state;
+		const { postAuthorId, instanceId, authors } = this.props;
+		const selectId = `post-author-selector-${ instanceId }`;
+		let selector;
+		if ( ! postAuthor ) {
+			return null;
+		}
 
-		/* eslint-disable jsx-a11y/no-onchange */
-		return (
-			<PostAuthorCheck>
-				<label htmlFor={ selectId }>{ __( 'Author' ) }</label>
+		if ( authors.length < 50 ) {
+			// Disable reason: A select with an onchange throws a warning.
+			/* eslint-disable jsx-a11y/no-onchange */
+			selector =
 				<select
 					id={ selectId }
-					value={ postAuthor }
-					onChange={ this.setAuthorId }
+					value={ postAuthorId }
 					className="editor-post-author__select"
+					onChange={ this.setAuthorId }
 				>
 					{ authors.map( ( author ) => (
 						<option key={ author.id } value={ author.id }>{ author.name }</option>
 					) ) }
-				</select>
+				</select>;
+		} else {
+			selector =
+				<Autocomplete
+					id={ selectId }
+					minLength={ 2 }
+					showAllValues={ true }
+					defaultValue={ postAuthor ? postAuthor.name : '' }
+					displayMenu="overlay"
+					onConfirm={ this.setAuthorId }
+					source={ this.suggestAuthor }
+					showNoResultsFound={ false }
+					tStatusQueryTooShort={ ( minQueryLength ) =>
+						// translators: %d: the number characters required to initiate an author search.
+						sprintf( __( 'Type in %d or more characters for results' ), minQueryLength )
+					}
+					tStatusNoResults={ () => __( 'No search results' ) }
+					tStatusSelectedOption={ ( selectedOption, length ) => __( `${ selectedOption } (1 of ${ length }) is selected` ) }
```

Comment:
I believe this will throw an `Translate function arguments must be string literals` eslint error when attempting to merge master into this branch

https://github.com/WordPress/gutenberg/pull/15877

Reply:
Good point @gwwr - fixed in https://github.com/WordPress/gutenberg/pull/7385/commits/b8300221f9cba9ded28096a7efd00cb031d88b04